ui/lib/pki/addon/components/pki-role-form.hbs

```diff
@@ -0,0 +1,93 @@
+<PageHeader as |p|>
+  <p.top>
+    <KeyValueHeader
+      @root={{hash label="role" text="role" path="vault.cluster.secrets.backend.pki.roles.index"}}
+      @isEngine={{true}}
+    >
+      <li>
+        <span class="sep">
+          /
+        </span>
+        <LinkTo @route="roles.index">
+          {{@model.backend}}
+        </LinkTo>
+      </li>
+    </KeyValueHeader>
+  </p.top>
+  <p.levelLeft>
+    <h1 class="title is-3">
+      {{#if @model.isNew}}
+        Create a PKI role
+      {{else}}
+        Edit a
+        {{@model.id}}
+      {{/if}}
+    </h1>
+  </p.levelLeft>
+</PageHeader>
+
+<form {{on "submit" (perform this.save)}}>
+  <div class="box is-sideless is-fullwidth is-marginless">
+    <MessageError @errorMessage={{this.errorBanner}} class="has-top-margin-s" />
+    {{! ARG TODO write a test for namespace reminder }}
+    <NamespaceReminder @mode={{if @model.isNew "create" "update"}} @noun="PKI role" />
+    {{! This is the code from FormFieldGroupsLoop. However, because we need to pass the attr into the yielded component, I'm unable to use that component.}}
```

Comment:
I'm not sure this comment is necessary as we've used this pattern of iterating over FormFields outside of the loop component

Reply:
I guess I just wanted to make a comment in case a future developer came along and noticed the code was nearly identical to the FormFieldGroupsLoop component and wondered why I didn't just use that component instead.

Reply:
Thank you! (reviewed over zoom), but passing through the attr in the yield the way you showed was awesome. A lot cleaner now.